src/jshal.js

```diff
@@ -73,35 +88,99 @@ mergeInto(LibraryManager.library, {
     },
 
     mp_js_hal_stdout_tx_strn: function(ptr, len) {
-        for (var i = 0; i < len; ++i) {
-            var c = String.fromCharCode(getValue(ptr + i, 'i8'));
-            term.write(c);
-        }
+        const data = UTF8ToString(ptr, len);
+        window.parent.postMessage({
+            kind: "serial_output",
+            data
+        }, "*");
+    },
+
+    mp_js_hal_temperature: function() {
+        return board.temperature.value;
+    },
+
+    mp_js_hal_button_get_presses: function(button) {
+        return board.buttons[button].getAndClearPresses()
+    },
+
+    mp_js_hal_button_is_pressed: function(button) {
+        return board.buttons[button].isPressed();
+    },
+
+    mp_js_hal_display_get_pixel: function(x, y) {
+        return board.display.getPixel(x, y);
     },
 
     mp_js_hal_display_set_pixel: function(x, y, value) {
-        display_context.fillStyle = `rgb(${value}, 0, 0)`;
-        display_context.fillRect(40 * x, 40 * y, 40, 40);
+        board.display.setPixel(x, y, value);
     },
 
-    mp_js_hal_audio_period_us: function(period_us) {
-        audio_frequency = 1000000 / period_us;
-        if (audio_osc) {
-            audio_osc.frequency.value = audio_frequency;
+    mp_js_hal_display_clear: function() {
+        board.display.clear();
+    },
+
+    mp_js_hal_display_read_light_level: function() {
+        return board.display.lightLevel.value;
+    },
+
+    mp_js_hal_accelerometer_get_x: function() {
+        return board.accelerometer.x.value;
+    },
+
+    mp_js_hal_accelerometer_get_y: function() {
+        return board.accelerometer.y.value;
+    },
+
+    mp_js_hal_accelerometer_get_z: function() {
+        return board.accelerometer.z.value;
+    },
+
+    mp_js_hal_accelerometer_get_gesture: function() {
+        // Equivalent to gesture_name_map.
+        // Is there a way to access e.g. MICROBIT_HAL_ACCELEROMETER_EVT_NONE ?
```

Comment:
I can't find a way. I've copied the constant #defines wholesale into board.js (will split out if/when we have bundling).

a18c4d82ce8db3de640bdf88ec4fef593e7af319